README.md

```diff
@@ -18,38 +18,283 @@ sources).
+    bmaptool: info: copying image 'tb-minimal-image-genericx86-64.rootfs.wic.gz' to block device '/dev/sdx' using bmap file 'tb-minimal-image-genericx86-64.rootfs.wic.bmap'
+    bmaptool: info: 100% copied
+    bmaptool: info: synchronizing '/dev/sdx'
+    bmaptool: info: copying time: 19.3s, copying speed 14.9 MiB/sec
+    ```
+
+## Booting
+
+To run TrenchBoot connect drive with flashed image to target platform and boot
+from it. In GRUB menu you can choose normal `boot` or `skl-boot`.
+
+```text
+                             GNU GRUB  version 2.06
+
+ +----------------------------------------------------------------------------+
+ |*boot                                                                       |
+ | skl-boot                                                                   |
+ |                                                                            |
+ |                                                                            |
+ |                                                                            |
+ +----------------------------------------------------------------------------+
+
+      Use the ^ and v keys to select which entry is highlighted.
+      Press enter to boot the selected OS, `e' to edit the commands
 ```
 
-Example output
+After a while you should see login prompt.
+
+```text
+early console in extract_kernel
+input_data: 0x0000000006801548
+input_len: 0x000000000121e953
+output: 0x0000000004600000
+output_len: 0x00000000033caee8
+kernel_total_size: 0x0000000003030000
+needed_size: 0x0000000003400000
+trampoline_32bit: 0x0000000000000000
+Physical KASLR using RDRAND RDTSC...
+Virtual KASLR using RDRAND RDTSC...
 
+Decompressing Linux... Parsing ELF... Performing relocations... done.
+Booting the kernel (entry_offset: 0x0000000000000000).
+
+
+Reference Yocto distro for PC Engines hardware 0.2.0 tb ttyS0
+
+tb login:
 ```
-# bmaptool copy --bmap tb-minimal-image-pcengines-apu2.wic.bmap tb-minimal-image-pcengines-apu2.wic.gz /dev/sda
-bmaptool: info: block map format version 2.0
-bmaptool: info: 3163136 blocks of size 4096 (12.1 GiB), mapped 587526 blocks (2.2 GiB or 18.6%)
-bmaptool: info: copying image 'tb-minimal-image-pcengines-apu2.wic.gz' to block device '/dev/sda' using bmap file 'tb-minimal-image-pcengines-apu2.wic.bmap'
-bmaptool: info: 97% copied
-bmaptool: info: 100% copied
-bmaptool: info: synchronizing '/dev/sda'
-bmaptool: info: copying time: 5m 30.3s, copying speed 6.9 MiB/sec
+
+To login use `root` account without password.
+
+## Running in QEMU
+
+It's possible to test image by running it in QEMU. Depending on QEMU
+configuration not all features may be available, SKL boot among others.
+
+To create bootable image you can once again use `bmaptool` but this time instead
+of writing image to flash device you write it to file
+
+```shell
+bmaptool copy --bmap tb-minimal-image-genericx86-64.rootfs.wic.bmap \
+        tb-minimal-image-genericx86-64.rootfs.wic.gz trenchboot.img
```

Comment:
I'm getting `bmaptool: error: unrecognized arguments: trenchboot.img`, but it works when I skip `--bmap` together with its file, it is detected automatically. This may be caused by too old `bmaptool` (tested on 3.5), I can try something newer tomorrow if it helps, but since bmap is autodetected anyway I think it should be skipped to make the command simpler.

Reply:
I used `bmaptool 3.6`. Works without `--bmap`, deleted in [9e7cf75](https://github.com/3mdeb/meta-trenchboot/pull/28/commits/9e7cf75d2df665cc8bb7e543765bdcacb9a094f4).